athena-google-bigquery/src/test/java/com/amazonaws/athena/connectors/google/bigquery/BigQueryTestUtils.java

```diff
@@ -137,4 +142,62 @@ static FieldValueList getBigQueryFieldValueList(Boolean bool, Integer integer, S
         return FieldValueList.of(generateBigQueryRowValue(bool, integer, string, floatVal),
                 FieldList.of(getTestSchemaFields()));
     }
+
+    static FieldValueList getBigQueryComplexTypeFieldValueList(List<Integer> values, Integer structFieldValue)
+    {
+        return FieldValueList.of(generateBigQueryRowValueForComplexTypes(values, structFieldValue),
+                FieldList.of(getComplexTypeTestSchemaFields()));
+    }
+
+    static List<Field> getComplexTypeTestSchemaFields()
+    {
+        return ImmutableList.of(
+                Field.newBuilder(LIST_FIELD_NAME_1, LegacySQLTypeName.INTEGER)
+                        .setMode(Field.Mode.REPEATED).build(),
+                Field.newBuilder(STRUCT_FIELD_NAME_1, LegacySQLTypeName.RECORD,
+                                FieldList.of(ImmutableList.of(Field.of(STRUCT_INT_FIELD_NAME_1, LegacySQLTypeName.INTEGER))))
+                        .setMode(Field.Mode.REPEATED).build()
+        );
+    }
+
+    static Schema getComplextTypeTestSchema()
+    {
+        return Schema.of(getComplexTypeTestSchemaFields());
+    }
+
+    static org.apache.arrow.vector.types.pojo.Schema getComplexTypeTestSchema()
+    {
+        return SchemaBuilder.newBuilder()
+                .addListField(LIST_FIELD_NAME_1, Types.MinorType.INT.getType())
+                .addStructField(STRUCT_FIELD_NAME_1).addChildField(STRUCT_FIELD_NAME_1,
+                        org.apache.arrow.vector.types.pojo.Field.nullable(STRUCT_INT_FIELD_NAME_1, Types.MinorType.INT.getType()))
+                .build();
+    }
+
+    static List<FieldValue> generateBigQueryRowValueForComplexTypes(List<Integer> intList, Integer structFieldValue)
+    {
+        List<FieldValue> values = new ArrayList<>();
+        intList.forEach(i -> values.add(FieldValue.of(FieldValue.Attribute.PRIMITIVE, intList == null ? null : String.valueOf(i))));
+        return ImmutableList.of(
+                FieldValue.of(FieldValue.Attribute.REPEATED, values),
+                FieldValue.of(FieldValue.Attribute.REPEATED,
+                         FieldValueList.of(generateStructRowValue(structFieldValue), FieldList.of(getStructField())))
+        );
+    }
+
+    static List<FieldValue> generateStructRowValue(Integer integer)
```

Comment:
Once again may not be following but why is `generateStructRowValue` returning List of Primitive `FieldValue`'s?